### DISCUSSION TITLE
Maximum call stack size exceeded error only on ios, React Vite

### DISCUSSION THREAD

**Discussion by KhomenkoSerhii** [community/community#128046] [Programming Help]
### Select Topic Area

Question

### Body

Hello! my application  works perfectly on the web, android/ios simulators and Android devices, but on iphones it gives an error on start page (Sign in page). It seems that the problem is in the redux store as it works without it.  Should i add any configs to vite.config or change store?

 "dependencies": {
 "@reduxjs/toolkit": "^2.0.1",
  "react-redux": "^9.1.0",
"vite": "^5.0.8",
  "react": "^18.2.0",
...
}
const reducers = { slice: ...};
const rootReducer = combineReducers(reducers);

const resettableRootReducer = (
  state: ReturnType<typeof rootReducer> | undefined,
  action: AnyAction,
) => {
  if (action.type === "store/reset") {
    return rootReducer(undefined, action);
  }
  return rootReducer(state, action);
};
const store = configureStore({
  reducer: resettableRootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }).concat(thunkMiddleware),
});
export default store;
export type AppDispatch = typeof store.dispatch;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
export const resetStore = () => {
  store.dispatch({ type: "store/reset" });
};
export type RootState = ReturnType<typeof store.getState>;

**Comment by richardfariax** [comment]
iOS-only `Maximum call stack size exceeded` with Redux + Vite often isn't a Vite config thing — it's usually **infinite render/dispatch** that Safari/WebKit on real devices catches more aggressively.

Things to check in your store setup:

1. **`resettableRootReducer` calling `rootReducer` inside itself** — make sure you're not accidentally recursing. Pattern should be:
```ts
const resettableRootReducer = (state, action) => {
  if (action.type === 'store/reset') {
    state = undefined
  }
  return rootReducer(state, action)
}
```
(not calling `rootReducer(undefined, action)` through another wrapper that bounces back)

2. **Selector creating new objects every render** → child re-renders → dispatch → loop. Use memoized selectors (`createSelector`) or pass primitives.

3. **Sign-in page `useEffect` dispatching on every render** because dependency array includes unstable refs.

Quick debug: temporarily log renders on SignIn. If count explodes before crash, it's a render loop not stack depth in Redux itself.

Also test on iOS **Safari WebView** vs installed PWA — if you're using Capacitor, same Redux code but different JS engine behavior.

Paste your `store.ts` and the SignIn component if you want a second pair of eyes.
